packages/aws-cdk/lib/api/util/rwlock.ts

```diff
@@ -156,9 +167,10 @@ async function readFileIfExists(filename: string): Promise<string | undefined> {
   }
 }
 
+let tmpCounter = 0;
 async function writeFileAtomic(filename: string, contents: string): Promise<void> {
   await fs.mkdir(path.dirname(filename), { recursive: true });
-  const tmpFile = `${filename}.${process.pid}`;
+  const tmpFile = `${filename}.${++tmpCounter}`;
```

Comment:
Needs both the counter and pid in here together methinks. Otherwise multiple processes might collide on the tmpcounter.

Reply:
The `filename` already includes the `pid` here... I'm relying on that there.